### ISSUE TITLE
Additive blending breaks scene transparency

### ISSUE THREAD

**Issue by gulagkulak** [obsproject/obs-studio#10681]
### Operating System Info

Other

### Other OS

Linux Mint 21.3

### OBS Studio Version

30.1.2

### OBS Studio Log URL

https://obsproject.com/logs/p6ENsxS0m3ho5ui7

### Expected Behavior

I built a 3D studio set in Blender and rendered out the background and foreground elements separately using a holdout matte, with the resulting foreground image having a transparent background. I have a separate glare/bloom pass for the foreground, which is supposed to be composited additively on top of the foreground layer.

To get clean glare/bloom out of Blender, the transparent background is replaced with a black background. In every other application (such as Gimp) compositing black on top of transparent additively results in transparency. The goal here is to have foreground elements throw bloom on top of the live host video.

### Current Behavior

Bringing these elements into OBS, the first solution I tried was using OBS folders/groups, expecting them to work like layer folders in Gimp where blending is limited only to the layer folder.

![image](https://github.com/obsproject/obs-studio/assets/144597589/2e85eb49-09d0-4d2a-a9d4-7dba3b6d72c2)

This didn't work as expected, however, as the foreground_fx image is additively blended with the background_combined layer, despite it being confined to its own folder/group, which blows out the colors of the background image, because it is already rendered with its own glare/bloom pass.

The second solution I tried was making a separate foreground scene with live video elements:

![image](https://github.com/obsproject/obs-studio/assets/144597589/6381ce11-d62c-4283-a3ab-457cc858da4d)

And then including that into another scene:

![image](https://github.com/obsproject/obs-studio/assets/144597589/26d42c6b-3afc-40f9-ae1f-71f4ea0d5fd3)

But this replaced all transparent areas of the foreground scene with pure black. The weird thing is that if I right-click on the foreground scene and select "Filters", under filters preview window it still shows the background as transparent (grey) instead of black.

**TLDR: What's the issue here?**

Having a subscene with transparency and a black-background FX image additively blended on top of it breaks scene transparency.

**What's the ideal solution here?**

Having the option to limit blending only to group/folder like Gimp does.

![image](https://github.com/obsproject/obs-studio/assets/144597589/0de997d6-7f74-49dc-8dcf-d3b1335259ad)

P.S: If you add a feature to OBS before the end of the month that limits blending only to group/folder and prevents additive blending from breaking transparency, I'll donate 20 euros to your ko-fi / patreon / favorite animal shelter / whatever.

### Steps to Reproduce

You can try compositing this test image additively on top of any image with transparency for a subtle bloom effect.

![image](https://github.com/obsproject/obs-studio/assets/144597589/277c84e2-8a64-415c-8e54-e5618775204f)

When a scene with this fx image and another image below it with transparency is included as a subscene in another scene, it replaces transparency with black.

**Comment by SuslikV** [comment]
Reason.
If you blend something not transparent to background it turns to not transparent area.

Not an issue.

But the scene's Filter preview is bugged (at least many moons ago it surely was, and it had OS independent behavior - I cannot test against v30.1.2, sorry).
For **scene's** Filter preview OBS blends sources to the window's colors when blending mode is set. Easy noticeable on single _Color Source_ of #ff0000 in the scene and _Blending Mode: Multiply_. When you select _Dark_ theme or _System_ theme - there background of the viewport's window is brighter, and against this color the source will be blended. It is no good if result of the Filter's preview depends on UI theme. When _Yami_ theme was introduced this bug was masked, because it was hard to distinguish the preview background from the window background.
